lib/core/text/flat.nit

```diff
@@ -922,9 +922,9 @@ class FlatBuffer
 
 	redef fun clear do
 		is_dirty = true
-		if written then reset
 		_bytelen = 0
 		_length = 0
+		if written then reset
```

Comment:
Basically reset is used when a buffer was to_s'd and modified afterwards.
In this case however, since we reset the buffer, resetting its values to 0 implies the buffer is re-allocated if necessary, but no amount of copy is needed, hence the modification here.